doc/ref_arch/openstack/chapters/chapter07.md

```diff
@@ -17,21 +19,70 @@
 
 <a name="7.1"></a>
 ## 7.1 Introduction
+To create an Infrastructure as a Service (IaaS) cloud requires the provisioning and deployment of the underlying infrastructure (compute, networking and storage) and deployment, configuration and management of the necessary software on the infrastructure; in the process of deploying the software, configuration of the infrastructure may also need to be performed. 
+
+Instead of deploying the infrastructure components and services manually, the current best practice is to write *code* (infrastructure as a Code, IaC) to define, provision, deploy, configure and manage the IaaS cloud infrastructure and services. IaC tools allows the entire provisioning, configuration and management processes to be automated. The desired state of the infrastructure and services is represented in a set of human readable, machine executable, and version-controlled files. With version control, it is easy to roll back to an older version and have access to the history of all committed changes.
+
+The provisioning of the infrastructure is typically performed by provisioning tools while the deployment of the software and the configuration of the software, and where needed the infrastructure, falls in the domain of configuration management tools. A single tool may support both provisioning and configuration management.
+
+Operators may choose certain paradigms with respect to how they provision and configure their IaaS cloud. These paradigms will drive the selection of the provisioning and configuration tools. In this chapter we will discuss the capabilities of provisioning and configuration management systems; some open source tools may be mentioned but their capabilities are beyond the scope of this chapter. 
+
+##### Procedural versus Declarative code
```

Comment:
Do we need these sections ```Procedural versus Declarative code``` and ```Mutable versus Immutable infrastructure``` as a part of this RA document. I don't think this should go in.

Reply:
@Pankaj I agree with @msidana about the 2 § "Procedural versus Declarative code" and "Mutable versus immutable infrastructure". In my opinion, these 2 paragraphs could be located in Reference model Chapter9. The notion is incompletely addressed in RM, and there are generic considerations common to all architectures

Reply:
@msidana @karinesevilla Since the content doesn't exist anywhere else and the concepts are referenced to in the following sections, they are needed. If I delete, questions would be asked about what is meant by this or that.

May I suggest that we accept as is and then someone can open an Issue against RM for the content to be moved.

Reply:
@pgoyal01 I agree to do that and create an issue against RM to move this content after it is merged in RA1

Reply:
Ok. Sounds good.